### PR TITLE
[MIR] Change SimplifyCfg pass to use BitVec

### PR DESCRIPTION
**Pull request by nagisa**
BitVector is more space efficient.

### CODE REVIEW

**Comment by bluss** [comment]
This was briefly discussed here https://github.com/rust-lang/rust/pull/30238#discussion_r46848632, merely some guesses on how big the dataset really is, and what's the appropriate fix. BitVec seems better than Vec either way.

**Comment by nagisa** [comment]
Rebased. This is a pretty simple one and independent from others so could land right now.

r? @nikomatsakis

File: src/librustc_data_structures/bitvec.rs [hunk 1]
```diff
@@ -50,6 +50,45 @@ impl BitVector {
...
+    iter: ::std::slice::Iter<'a, u64>,
+    current: u64,
+    idx: usize
+}
+
+impl<'a> Iterator for BitVectorIter<'a> {
+    type Item = usize;
+    fn next(&mut self) -> Option<usize> {
+        while self.current == 0 {
+            self.current = if let Some(&i) = self.iter.next() {
```
Comment (nikomatsakis) [inline src/librustc_data_structures/bitvec.rs]: It seems like it'd be easier/clearer to just keep a slice with some indices here.

Comment (nikomatsakis) [inline src/librustc_data_structures/bitvec.rs]: (But don't feel you have to change it.)

**Pull request merged**